crates/ra_hir/src/expr.rs

```diff
@@ -792,40 +795,79 @@ impl ExprCollector {
             ast::ExprKind::Label(_e) => self.alloc_expr(Expr::Missing, syntax_ptr),
             ast::ExprKind::IndexExpr(_e) => self.alloc_expr(Expr::Missing, syntax_ptr),
             ast::ExprKind::RangeExpr(_e) => self.alloc_expr(Expr::Missing, syntax_ptr),
-            ast::ExprKind::MacroCall(_e) => self.alloc_expr(Expr::Missing, syntax_ptr),
+            ast::ExprKind::MacroCall(e) => {
+                let name = e.name().map(|nr| nr.as_name()).unwrap_or_else(Name::missing);
+
+                if let Some((module, _)) = self.resolver.module() {
+                    let def = module.find_macro(&name);
+
+                    if let Some(def_id) = def {
+                        let rules = db.macro_def(*def_id);
+
+                        if let Some(rules) = rules {
+                            let tt = e.token_tree().map(mbe::ast_to_token_tree);
+
+                            if let Some(Some((tt, _))) = tt {
+                                let result = rules.expand(&tt);
+
+                                if let Ok(tt) = result {
+                                    // let items = ;
+
+                                    mbe::token_tree_to_ast_item_list(&tt)
+                                        .syntax()
+                                        .descendants()
+                                        .find_map(ast::Block::cast)
```

Comment:
Instead of using the syntax().descendants() I could iter through the modules and the impl items. Converting them to the right `Function`, `Const` or `Static` struct and then excute the hir_parse query